code/__DEFINES/jobs.dm

```diff
@@ -418,3 +418,84 @@
 		if(jobname in hud_to_department_acc)
 			return hud_to_department_acc[jobname]
 	return ACCOUNT_CIV
+
+// used to determine chat color by HUD in `chatmessage.dm`
+// Note: custom colors are what I really didn't put much attention into. feel free to change its color when you feel off.
+/datum/chatmessage/proc/get_chatcolor_by_hud(jobname)
+	if(jobname)
+		var/static/hud_to_chatcolor = list(
```

Comment:
I think it's cleaner if the colour is stored in the job datum and this list is generated based on the initial values of the job datums

Reply:
that would work. I will try it.

Reply:
okay, I thought it would work, and in my simulation, it becomes more messy because `SSjob.GetJob()` should be called by a key with the correct job name. `HUD_JOB_SOMETHING` is all lower cases, and calling them need a proc that gets a job title when it's given `HUD_JOB_SOMETHING`.
i.e.) If a proc takes `headofsecurity`, it should return `Head of Security`, and `Head of Security` is a key to call a job datum.

Reply:
It's not cleaner, there's no need to store it in useless, never constant variable.
Storing it in one global list (with some sort of mapper proc) is cleaner and easier to find.